.travis.yml

```diff
@@ -1,30 +1,50 @@
-sudo : true
-dist : trusty
+os: linux
 language: node_js
-node_js:
-  - "node"
+node_js: "14"
+
 addons:
-  sauce_connect: true
   ssh_known_hosts: download.kiwix.org
-before_script:
-    - npm install nightwatch http-server
-    - "./node_modules/http-server/bin/http-server . &"
-script:
-    # The free account on Sauce does not allow more than 5 concurrent sessions (including the main one)
-    # So we separate the recent and old browsers in order to respect this limit.
-    # NB : latest edge should be added to this list when Microsoft releases the new Edge based on Chromium
-    - "./node_modules/nightwatch/bin/nightwatch -c nightwatch.js --env firefox,chrome"
-    - "./node_modules/nightwatch/bin/nightwatch -c nightwatch.js --env edge40,edge44"
-    - "./node_modules/nightwatch/bin/nightwatch -c nightwatch.js --env firefox45,chrome58,ie11"
-    - pkill node || echo "Node process not running (anymore)"
-deploy:
-    - provider: script
-      # Nightly builds (launched by cron)
-      script: sudo apt-get update -q && sudo apt-get --yes install click click-reviewers-tools && ./scripts/setup_travis_env.sh && DISPLAY=:99.0 ./scripts/create_all_packages.sh
-      on:
-          condition: ( "$TRAVIS_EVENT_TYPE" = "cron" )
-    - provider: script
-      # Generation of packages for public releases (launched by a tag)
-      script: sudo apt-get update -q && sudo apt-get --yes install click click-reviewers-tools && ./scripts/setup_travis_env.sh && DISPLAY=:99.0 ./scripts/create_all_packages.sh -t -v ${TRAVIS_TAG}
-      on:
-          tags: true
+
+jobs:
+  fast_finish: true
+  include:
+
+    - stage: "Local testing"
+      node_js: "10"
+      addons:
+        firefox: latest-esr
+        chrome: stable
+      script: npm test
+
+    - stage: "Cross-browser testing"
+      # This stage only runs when saucelabs credentials are available
+      # (e.g. direct pushes and local non-fork PRs)
+      # https://docs.travis-ci.com/user/environment-variables/
+      # https://docs.travis-ci.com/user/conditions-v1
+      if: NOT fork
+      addons:
+        sauce_connect: true
+      script:
+        - npm run test-unit-saucelabs
+        - "./node_modules/.bin/http-server . &"
+        - sleep 2
+        - curl "http://localhost:8080" | head
+        # The free account on Sauce does not allow more than 5 concurrent sessions (including the main one)
+        # So we separate the recent and old browsers in order to respect this limit.
+        # NB : latest edge should be added to this list when Microsoft releases the new Edge based on Chromium
```

Comment:
Edge Chromium has now been out for a while -- maybe we can add edge87 which I think is the current stable version (it is my version which is a standard install, not a dev channel). If edge87 is not available, then edge86.

Reply:
Ah, OK! Fine by me. There are some specifics added to Edge, though always rebased on the Chromium codebase, and it is often released behind the equivalent Chrome, so probably still worth adding to tests -- especailly since we now target the Edge Add-in store for the Chromium extension. But only if it doesn't cause complications.

Reply:
I didn't want to mix nominal changes with the refactor. Happy to push it separately or here, whichever you prefer :)

Reply:
@Krinkle I don't want to alter your working practices, so whatever is easiest for you.

Reply:
If edge chromium is included then this pr would address #502 as well.